modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java

```diff
@@ -2164,13 +2256,66 @@ private boolean rebuildIsMeaningless(GridCacheContext cctx) {
         }
 
         try {
-            return idx.rebuildIndexesFromHash(cctx);
+            return rebuildIndexesFromHash0(cctx, false);
```

Comment:
This the only place where `rebuildInMemory=false`.
It seems we can eliminate this new flag in the bellow methods if just add if here
```
            if (cctx.group().persistenceEnabled())
                return rebuildIndexesFromHash0(cctx);
```

Also we can just remove persistence check in the 
```
        // No data in fresh in-memory cache. If force, this meant that indexing is enabling dynamically.
        if (!cctx.group().persistenceEnabled() && !rebuildInMemory)
            return null;
```

Reply:
It seems that this check is also redundant, fixed.